src/ZendeskApi.Client/Resources/AbstractBaseResource.cs

```diff
@@ -0,0 +1,211 @@
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using ZendeskApi.Client.Extensions;
+using ZendeskApi.Client.Formatters;
+using ZendeskApi.Client.Models;
+
+namespace ZendeskApi.Client.Resources
+{
+    public abstract class AbstractBaseResource<TResource>
+    {
+        protected readonly IZendeskApiClient ApiClient;
+        protected readonly ILogger Logger;
+        protected readonly string DocsResource;
+
+        protected readonly Func<ILogger, string, IDisposable> LoggerScope = LoggerMessage.DefineScope<string>(typeof(TResource).Name + ": {0}");
+
+        protected AbstractBaseResource(
+            IZendeskApiClient apiClient, 
+            ILogger logger,
+            string docsResource)
+        {
+            ApiClient = apiClient;
+            Logger = logger;
+            DocsResource = docsResource;
+        }
+
+        protected async Task<T> GetAsync<T>(
+            string resource,
+            string docs,
+            string scope,
```

Comment:
Can we provide a default scope here? As we have done in the `CreateAsync` method?

Reply:
In the three examples I looked at the scope was different for almost every usage so it didn't make sense to provide a default here.

Reply:
The should be three resource examples but each resource calls this base method multiple times.